x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/create_query_alert_type.ts

```diff
@@ -97,6 +101,16 @@ export const createQueryAlertType = (
         primaryTimestamp,
         secondaryTimestamp,
       });
+
+      if (responseActions?.length && result.createdSignalsCount) {
+        scheduleNotificationResponseActions(
+          {
+            signals: result.createdSignals,
+            responseActions: responseActions as unknown as ResponseAction[],
```

Comment:
This is quite a dangerous cast, as there's no validation in the HTTP API schema or internal detection engine rule schemas to verify that `responseActions` has this type. We should have stricter types for the `responseAction` params, which will add the validation and remove the need for this cast.

Reply:
What type do you suggest? So far it has `actionTypeId: string` and `params: { [p: string]: unknown }`. I think that all the params are Record<string, any>.

Reply:
I would suggest updating the `ResponseAction` io-ts type so the `params` are not `Record<string, any>` but instead validate the expected params when the `action_type_id` is `.osquery`. Something like 
```
const ResponseAction = t.exact(
  t.type({
    action_type_id: t.literal('.osquery'),
    params: // specific io-ts object for osquery params
  })
);
```
When we want to add other types of response actions, we can create a union of io-ts types that have different literals for `action_type_id`.

Similarly, the server response action io-ts type (ResponseActionRuleParam) should have specific validation for `params`. The extra interface `OsqueryResponseAction` and logic to cast to the interface should then be unnecessary. Right now the lack of io-ts validation will allow the API to accept anything in the params (Record<string, any>) so some params could be missing or invalid.